server/routes/activities/manage-allocations/handlers/endDate.ts

```diff
@@ -51,7 +53,26 @@ export class EndDate {
 }
 
 export default class EndDateRoutes {
-  GET = async (req: Request, res: Response) => res.render('pages/activities/manage-allocations/end-date')
+  GET = async (req: Request, res: Response): Promise<void> => {
+    const { allocateJourney } = req.session
+    const nextAvailableInstance = allocateJourney.scheduledInstance
+    const nextSessionDateAndTime = parseDate(
+      `${nextAvailableInstance.date}T${nextAvailableInstance.startTime}`,
+      "yyyy-MM-dd'T'HH:mm",
+    )
+
+    if (
+      config.deallocateTodaySessionEnabled &&
+      allocateJourney.inmates.length === 1 &&
+      isToday(nextSessionDateAndTime) &&
+      !isPast(nextSessionDateAndTime) &&
+      !allocateJourney.deallocateTodayOption
```

Comment:
If the following is true then user can select to de-allocate from sessions later today:

- feature toggle is enabled
- only one prisoner is selected
- the next session is today
- the next session is later today
- user has not chosen to de-allocate today already